src/v/redpanda/debug_bundle.cc

```diff
@@ -0,0 +1,510 @@
+    // First check if the requested filename is actually a bundle
+    // TODO(@NyaliaLui): In the future when we support more than 1 bundle on
+    // disk, this condition will evolve to check if the filename follows our
+    // naming scheme.
+    if (filename != default_bundle_name) {
+        co_return errc::invalid_filename;
+    }
+
+    // Then check if the requested filename is on disk
+    auto exists = co_await ss::file_exists(
+      make_bundle_filename(get_write_dir(), filename));
+    auto core0_running = co_await is_core0_running();
+    if (!exists) {
+        if (core0_running) {
+            co_return errc::bundle_not_ready;
+        } else {
+            co_return errc::bundle_not_on_disk;
+        }
+    } else {
+        // It is possible for the status check to occur in a small window where
+        // the spawned process is still running but RPK has written to disk
+        // already.
+        if (core0_running) {
+            co_return errc::bundle_not_ready;
+        }
+    }
+
+    co_return errc::success;
+}
+
+ss::future<std::error_code> debug_bundle::remove_bundle(ss::sstring filename) {
+    gate_guard guard{_rpk_gate};
+    auto units = co_await _rpk_mu.get_units();
+
+    auto err = co_await is_ready(filename);
+    if (make_rpk_err_value(err) != errc::success) {
+        co_return err;
+    }
+
+    co_return co_await container().invoke_on(
+      debug_bundle_shard_id, [&filename](debug_bundle& b) {
+          // rpk.do_remove will acquire the mutex that protects spawned
+          // processes. This is to protect against situations where a new
+          // process is spawned before remove is called.
+          return b._rpk->do_remove(filename);
+      });
+}
+
+ss::future<std::error_code> debug_bundle::do_abort(bool use_sigkill) {
+    if (_rpk == nullptr || _rpk->process == nullptr) {
+        co_return errc::nothing_running;
+    }
+
+    co_return _rpk->do_abort(use_sigkill);
+}
+
+ss::future<std::error_code> force_stop(debug_bundle& b) {
+    // If process is running
+    // do SIGTERM
+    // WAIT
+    // If process is still running (process is stopped at the end of
+    // rpk_debug_bundle::handle_process()) otherwise, do SIGKILL
+
+    auto rpk_running = co_await b.is_rpk_running();
+    if (!rpk_running) {
+        // First check returns nothing_running because the abort is unecessary
+        co_return errc::nothing_running;
+    }
+
+    co_await b.do_abort(false);
+
+    co_await ss::sleep(rpk_abort_sleep_ms);
+
+    rpk_running = co_await b.is_rpk_running();
+    if (!rpk_running) {
+        // Second check returns success because the abort worked
+        co_return errc::success;
+    }
+
+    co_await b.do_abort(true);
+
+    // Do not reset _rpk here, only debug_bundle::start_creating_bundle() can do
+    // that
+    // TODO(@NyaliaLui): make a Singleton class for _rpk with
+    // start_creating_bundle() as the only friend
```

Comment:
@michael-redpanda This comment is what I was referring to in the slack message. Do you want to tackle this and the other list as well async?